src/endpoints/sources.js

```diff
@@ -305,40 +342,72 @@ function sendSubresource(subresourceExtractor) {
 
     authz.assertAuthorized(req.user, authz.actions.Read, subresource.resource);
 
-    const subresourceUrl = await subresource.url();
+    return await proxyFromUpstream(req, res,
+      await subresource.url(),
+      subresource.accept
+    );
+  };
+}
 
-    /* Proxy the data through us:
-     *
-     *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
-     */
-    return await proxyResponseBodyFromUpstream(req, res, new Request(subresourceUrl, {
-      headers: {
-        Accept: subresource.accept,
-
-        /* Normalize Accept-Encoding to gzip or nothing, as realistically we
-         * only expect upstreams to support gzip and normalizing reduces the
-         * cache burden of varying on many equivalent but effectively the same
-         * client Accept-Encoding values.
-         */
-        "Accept-Encoding": req.acceptsEncodings("gzip")
-          ? "gzip"
-          : "identity",
-      },
-
-      /* Use "no-cache" mode to always revalidate with the upstream but avoid
-       * re-transferring the content if we have a cached copy that matches
-       * upstream.
-       */
-      cache: "no-cache",
 
-      /* Disable automatic body decompression.  Instead, we pass through the
-       * Accept-Encoding of the client to the upstream and return the
-       * upstream Content-Encoding and body directly.  This avoids needlessly
-       * decompressing and recompressing ourselves.
+/**
+ * Proxy the data through us:
+ *
+ *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
+ *
+ * @param {express.request} req - Express-style request instance
+ * @param {express.response} res - Express-style response instance
+ * @param {String} url - Resource URL
+ * @param {String} accept - HTTP Accept header value
+ * @returns {expressEndpointAsync}
+ */
+async function proxyFromUpstream(req, res, url, accept) {
+  return await proxyResponseBodyFromUpstream(req, res, new Request(url, {
```

Comment:
`@returns {expressEndpointAsync}` isn't right here since it's effectively:

https://github.com/nextstrain/nextstrain.org/blob/2adabdb426743f2c4b254ceb2cbc7bcbe1e9ec6b/src/endpoints/sources.js#L632

Reply:
Fixed: ecc6916b0db78d4066aad838ed8c6ceed75b72d0, aaaf67cb9335ead33ee44f690a90d63f6c706259